poc-data-api/Extracting_Water_Temperature_at_Site.Rmd

```diff
@@ -0,0 +1,251 @@
+
+```{r lat_lon}
+data_df %>% 
+  #We select unique sites included in the dataset together with lat and lon values
+  distinct(site, subsite, lon, lat) %>% 
+  #Selecting the first rows for comparison with option#1
+  arrange(site) %>% 
+  head() %>% 
+  #We load them into memory
+  collect()
+```
+The two options shown above for extracting latitude and longitude values for each site are equivalent and produce the same results. They are interchangeable and you can use whichever you prefer.
+
+
+# Plotting map of sampled sites in this dataset
+
+We will make a map of all sites included in the AIMS Water Temperature Monitoring Program. We will also highlight the location of two sites: **Hayman Island** and **Heron Island** in red. We will use these highlighted sites as examples to show you how to extract data at multiple sites of interest.
+
+```{r map_sites}
+#Loading basemap with countries in Oceania and Asia
+oce_asia <- rnaturalearth::ne_countries(continent = c("oceania", "asia"), 
+                                     returnclass = "sf")
+
+#Creating map with all sites included in dataset
+#First we plot the basemap with the countries in Oceania and Asia
+oce_asia %>% 
+  ggplot()+
+  geom_sf()+
+  #Plotting monitoring sites
+  geom_point(inherit.aes = F, data = sites, aes(x = lon, y = lat))+
+  geom_point(inherit.aes = F, data = sites[sites["site"] == "Hayman Island" | sites["site"] == "Heron Island",], 
+             aes(x = lon, y = lat), color = "red", size = 2)+
+  #Changing map limits - Adding a buffer of 2 degree around max and min site coordinates
+  lims(y = c(floor(min(sites$lat))-2, ceiling(max(sites$lat))+2),
+       x = c(floor(min(sites$lon))-2, ceiling(max(sites$lon))+2))+
+  #Removing default grey background
+  theme_bw()
+```
+
+# Extracting data for sites of interest
+
+We can use the information in our `sites` variable to extract data for a site of interest. All we need to know is the name of the site, as shown below.  
+  
+Here, we will use data collected at **Hayman Island** and **Heron Island** as an example. We will calculate monthly means at each site, so we can plot time series that will show us how temperature has changed over time.
+
+```{r site_coords}
+sites_coords <- sites %>% 
+  #Using random site as example
+  filter(site %in% c("Hayman Island", "Heron Island")) %>% 
+  #Extracting latitude and longitude coordinates
+  select(site, lon, lat, deployment_location)
+
+#Checking results
+sites_coords
+```
+
+We can see that there are multiple coordinates for each site, but they are located in close proximity to one another. In this case, both sites include instrument deployments in the reef flat and slope. We will use this information to extract data from the original AIMS dataset.  
+  
+Note that `qc_val` is the variable containing the temperature data in this dataset.

```

Comment:
Note that `qc_val` is the variable containing quality-controlled temperature data in this dataset.